core/services/ocr2/plugins/functions/reporting.go

```diff
@@ -330,6 +361,20 @@ func (r *functionsReporting) Report(ctx context.Context, ts types.ReportTimestam
 			"requestID":     reqId,
 			"nObservations": len(observations),
 		})
+		if r.contractVersion == 1 {
+			var requestCoordinator common.Address
+			requestCoordinator.SetBytes(aggregated.CoordinatorContract)
+			reportCoordinator, err = ShouldIncludeCoordinator(&requestCoordinator, reportCoordinator)
+			if err != nil {
+				r.logger.Error("FunctionsReporting Report: skipping request with mismatched coordinator contract address", commontypes.LogFields{
```

Comment:
This should only occur if their is a faulty leader which returns a query that contains requests from different coordinators, right?

Reply:
yes